packages/sdk/src/across/clients/bobaBridge.ts

```diff
@@ -0,0 +1,101 @@
+import assert from "assert";
+import { ethers } from "ethers";
+import { Provider } from "@ethersproject/providers";
+import { Signer, BigNumber, ContractTransaction } from "ethers";
+import {
+  ERC20Ethers__factory,
+  OptimismL1StandardBridgeEthers__factory,
+  BobaAddressManagerEthers__factory,
+} from "@uma/contracts-node";
+import { Watcher } from "@eth-optimism/core-utils";
+
+export const l1Contracts: { ADDRESS_MANAGER_ADDRESS: { [chainId: number]: string } } = {
+  ADDRESS_MANAGER_ADDRESS: {
+    // mainnet
+    1: "0x8376ac6C3f73a25Dd994E0b0669ca7ee0C02F089",
+    // rinkeby
+    4: "0x93A96D6A5beb1F661cf052722A1424CDDA3e9418",
+  },
+};
+
+export class BobaBridgeClient {
+  // Gas limit for the L2 transaction initiated by the Sequencer
+  public readonly L2_DEPOSIT_GAS_LIMIT = 1300000;
+
+  /**
+   * Create a transaction to deposit ERC20 tokens to Boba. Mainnet and Rinkeby are currently supported
+   * @param l1Signer The L1 wallet provider (signer)
+   * @param l1Erc20Address The L1 token address
+   * @param l2Erc20Address The L2 token address
+   * @param amount The amount to be deposited in wei
+   * @returns The submitted transaction
+   */
+  async depositERC20(l1Signer: Signer, l1Erc20Address: string, l2Erc20Address: string, amount: BigNumber) {
+    const chainId = await l1Signer.getChainId();
+    const addressManagerAddress = l1Contracts.ADDRESS_MANAGER_ADDRESS[chainId];
+    assert(typeof addressManagerAddress === "string", "Chain not supported");
+
+    const addressManager = BobaAddressManagerEthers__factory.connect(addressManagerAddress, l1Signer);
+    const l1StandardBridgeAddress = await addressManager.getAddress("Proxy__OVM_L1StandardBridge");
+    const l1StandardBridge = OptimismL1StandardBridgeEthers__factory.connect(l1StandardBridgeAddress, l1Signer);
+    const l1_ERC20 = ERC20Ethers__factory.connect(l1Erc20Address, l1Signer);
+    return l1StandardBridge.depositERC20(
```

Comment:
Do approvals need to happen before this? Should we offer a utility so users can 1) check if they've already approved the bridge to pull their tokens and 2) do the approval?

Reply:
Approvals are required, yes, but this logic already exists in the FE app for other contracts. I think that exporting the Bridge addresses is enough

Reply:
I added methods for approve and allowance just in case